src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py

```diff
@@ -26,5 +28,27 @@ def load_arguments(self, _):
         rs.argument('grain', options_list=['--grain', '-g'], type=str, help='Reservation summaries grain. Possible values are daily or monthly')
 
     with self.argument_context('consumption pricesheet show') as cps:
+        cps.argument('top', options_list=['--top', '-t'], type=int, help='maximum number of items to return. Accepted range for this value is 1 - 1000')
         cps.argument('include_meter_details', options_list=['--include-meter-details', '-m'], action='store_true', help='include meter details in the price sheet')
         cps.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the price sheet')
+
+    with self.argument_context('consumption marketplace list') as cmp:
+        cmp.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the marketplace')
+        cmp.argument('top', options_list=['--top', '-t'], type=int, help='maximum number of items to return. Accepted range for this value is 1 - 1000')
+        cmp.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+        cmp.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+
+    with self.argument_context('consumption budget') as cb:
+        cb.argument('resource_group_name', options_list=['--resource-group-name', '-r'], action='store_true', help='get budget list for subscription by specific resource group name')
+        cb.argument('budget_name', options_list=['--budget-name', '-b'], help='get budget information by budget name')
```

Comment:
This help text seems specific only to a show command but it would apply to a delete command as well.  It should be more generic. ("The budget name.") or different help text should be applied to different scopes.

Reply:
We plan to keep budget arguments in params, and updated type and help of some arguments